### PR TITLE
feat: shell for deploy and retrieve oas

### PR DESCRIPTION
**Pull request by CristiCanizales**
### What does this PR do?
- Creates structure for deploying/retrieving oas files

### What issues does this PR fix or reference?
@W-17341339@, @W-17341336@

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.json [hunk 1]
```diff
@@ -194,15 +194,15 @@
       "editor/context": [
         {
           "command": "sf.retrieve.current.source.file",
-          "when": "sf:project_opened && resourceLangId != 'forcesourcemanifest' && sf:has_target_org"
+          "when": "sf:project_opened && resourceLangId != 'forcesourcemanifest' && resourceLangId != 'yaml' && sf:has_target_org"
```
Comment (mingxuanzhangsfdx) [inline packages/salesforcedx-vscode-core/package.json:197]: From my understanding, you need to define the resourceLangId yaml [here](https://github.com/forcedotcom/salesforcedx-vscode/blob/feat/apex-oas/packages/salesforcedx-vscode-core/package.json#L1005) in order to use it.

Comment (CristiCanizales) [reply packages/salesforcedx-vscode-core/package.json:197]: I believe it's not needed. I tested it just like this and works fine

Comment (mingxuanzhangsfdx) [reply packages/salesforcedx-vscode-core/package.json:197]: Thanks for the confirmation!

File: packages/salesforcedx-vscode-core/package.json [hunk 2]
```diff
@@ -194,15 +194,15 @@
...
         },
         {
           "command": "sf.retrieve.in.manifest",
           "when": "sf:project_opened && resourceLangId == 'forcesourcemanifest' && sf:has_target_org"
         },
         {
           "command": "sf.deploy.current.source.file",
-          "when": "sf:project_opened && resourceLangId != 'forcesourcemanifest' && sf:has_target_org"
+          "when": "sf:project_opened && resourceLangId != 'forcesourcemanifest' && resourceLangId != 'yaml' && sf:has_target_org"
```
Comment (peternhale) [inline packages/salesforcedx-vscode-core/package.json:205]: @CristiCanizales I am starting to get the feeling that the introduction of a new deployable resource (yaml files) is starting to splinter the cohesion of the notion o deploy/retrieve.

Do we really need separate commands to do a deploy/retrieve for or should we use a single command and then let the command's impl dispatch the deploy/retrieve according to what was picked.

Comment (CristiCanizales) [reply packages/salesforcedx-vscode-core/package.json:205]: I don't think the user will tell the difference. It's like with manifest files, there's also a separate command to deploy their contents. User only sees one deploy command in each of the different categories (stuff in force-app, xml files and yml files) so there's no confusion and in the end they're in fact just deploying to org

Comment (peternhale) [reply packages/salesforcedx-vscode-core/package.json:205]: I understand your point and it has worked well for us. I am just wondering if there might be a way to reduce command proliferation.

Maybe not a good discussion to have now.

**Review by peternhale — APPROVED** [review]

**Pull request merged**